pkg/util/admission/granter.go

```diff
@@ -620,13 +614,14 @@ func (sg *kvStoreTokenGranter) tryGrant() {
 
 // tryGrantLocked attempts to grant to as many requests as possible.
 func (sg *kvStoreTokenGranter) tryGrantLocked() {
-	for sg.requesterHasWaitingRequests() && sg.tryGrantLockedOne() {
+	for sg.tryGrantLockedOne() {
```

Comment:
This seems like a behavior change, can you explain?

Reply:
Was this redundant because `kvStoreTokenGranter.tryGrantOneLocked` already calls `hasWaitingRequests`? But the semantics still don't seem identical, now we call `sg.(one-of-two-options).hasWaitingRequests` and before we returned true if `(any-of-three)` had waiting requests.